frontend_lib/src/appContentFactory.js

```diff
@@ -587,6 +709,96 @@ export function appContentFactory (WrappedComponent) {
       return sortTimelineByDate(fullTimeline)
     }
 
+    loadMoreTimelineItems = async (getContentRevision) => {
+      const { props, state } = this
+
+      const newItemCount = state.timeline.length + TIMELINE_ITEM_COUNT_PER_PAGE
+
+      if (state.wholeTimeline.length >= newItemCount) {
+        this.setState(prevState => {
+          return {
+            timeline: this.getTimeline(state.wholeTimeline, newItemCount)
+          }
+        })
+        return
+      }
+
+      const fetchResult = async (fetchPromise) => {
+        return await handleFetchResult(await fetchPromise)
+      }
+
+      // Get the newest comments, files and revisions from the paginated backend API
+      const [commentsResponse, filesResponse, revisionsResponse] = await Promise.all([
+        fetchResult(getContentComment(this.apiUrl, state.content.workspace_id, state.content.content_id, state.commentPageToken, TIMELINE_ITEM_COUNT_PER_PAGE, 'created:desc')),
+        fetchResult(getFileChildContent(this.apiUrl, state.content.workspace_id, state.content.content_id, state.filePageToken, TIMELINE_ITEM_COUNT_PER_PAGE, 'created:desc')),
+        // INFO - 2021/08/17 - S.G. - the order is done with "modified" for revisions as the "created" field is the creation
+        // date of the content, not of a revision (and thus is the same for all revisions of a content).
+        fetchResult(getContentRevision(this.apiUrl, state.content.workspace_id, state.content.content_id, state.revisionPageToken, TIMELINE_ITEM_COUNT_PER_PAGE, 'modified:desc'))
+      ])
+
+      if (!commentsResponse.apiResponse.ok && !filesResponse.apiResponse.ok && !revisionsResponse.apiResponse.ok) {
+        sendGlobalFlashMessage(props.t('Error while loading timeline'))
+        console.log('Error loading timeline', 'comments', commentsResponse, 'revisions', revisionsResponse, 'files', filesResponse)
```

Comment:
Let's make the error more visible in the console?

```suggestion
        console.error('Error loading timeline', 'comments:', commentsResponse, 'revisions:', revisionsResponse, 'files:', filesResponse)
```
(or is it a forgotten console.log?)

Reply:
Nope, I just moved this part.

Reply:
Your suggestion has been applied.